File: GoalMood.BE/Models/TeamMember.cs
namespace GoalMood.BE.Models;

/// <summary>
/// Represents a team member
/// </summary>
public class TeamMember
{
    /// <summary>
    /// Unique identifier
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Member name (1-50 characters)
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Current mood state
    /// </summary>
    public Mood CurrentMood { get; set; } = Mood.Neutral;

    /// <summary>
    /// Navigation property for goals (not mapped to DB in Dapper)
    /// </summary>
    public List<Goal>? Goals { get; set; }
}

File: GoalMood.FE/src/types/TeamMember.ts
import type { Goal } from './Goal'

export enum Mood {
  Happy = 1,
  Content = 2,
  Neutral = 3,
  Sad = 4,
  Stressed = 5
}

export interface TeamMember {
  id: number
  name: string
  currentMood: Mood
  moodEmoji: string
  goals: Goal[]
  completedCount: number
  totalCount: number
}
